src/controllers/router-store/index.tsx

```diff
@@ -140,7 +141,15 @@ const actions: AllRouterActions = {
 
           batch(() => {
             plugins.forEach(p => {
-              if (p.id === 'resources-plugin' || !sameRoute) {
+              if (
+                shouldReload({
+                  context: nextContext,
+                  prevContext,
+                  pluginId: p.id,
+                  defaultShouldReload:
```

Comment:
Can `defaultShouldReload` and `sameRoute` just move to the internals of `shouldReload` considering it receives the `pluginId`?

Reply:
You could avoid that if you pre-process the whole thing, but you would need one more iteration for each plugin (this is cheap, the shallow equal on the routes may be more expensive).

```
const shouldReloadByPlugin = new Map(
  plugins.map((plugin) =>
    [plugin.id, shouldReload({ context, prevContext, pluginId: plugin.id })]
  )
);

// example usage
for (const plugin of plugins) {
  if (shouldReloadByPlugin.get(plugin.id)) {}
}


```

Reply:
make sense, let me do it

Reply:
done